### DISCUSSION TITLE
htmx Todo MVC and Thymeleaf

### DISCUSSION THREAD

**Discussion by checketts** [wimdeblauwe/wimdeblauwe.com#1] [Q&A] [answered]
I'm getting very interested in htmx (along with Thymeleaf). One puzzling item I'm noticing is when checking a todo, it makes 2 calls to the `/active-items-count` endpoint. This seems to come from the `hx-trigger` but other than that, I'm not sure why it happens.

Secondly, I found the thymeleaf/htmx attributes hard to read (due to the `th:attr` merging a lot of things, is there any interest in creating an HtmxDialect to help with that?

**Comment by wimdeblauwe** [comment, marked as answer]
I noticed you figured out the first point: https://github.com/wimdeblauwe/blog-example-code/issues/7 so thank you for that!

 2nd: What a great idea! I am definitely going to try to put together a POC on that this week.

**Reply by checketts** [reply]
I would be happy to collaborate on that (though I'm still pretty new to Thymeleaf). I'll keep my eyes open if it come up.

**Reply by wimdeblauwe** [reply]
Super! What do you think would be a good prefix to use for the custom dialect? `thhx:` or maybe `htmx:`? Other ideas? I think `hx:` would be too confusing if you get `hx:hx-get` or `hx:get`.

**Reply by checketts** [reply]
I used `hx` in my sample one. That way `hx-target` might be used if the the value is static, then `hx:target` if Thymeleaf processing is needed. In both cases `hx` is a namespacing prefix. Also we could support overlapping ones like `hx:hx-get` if it comes up that it is a common confusion.

**Reply by wimdeblauwe** [reply]
Ah, you already have a sample one? Any way you can already share?

**Reply by checketts** [reply]
Sure. here is my experiment: https://github.com/checketts/wimdeblauwe.com/commit/fd701bf7b88b91e8f9b6eab99f89cfec71def634

Just playing in your todo-mvc repo so far. I'm happy to move it to a new repository to explore independently.

**Reply by wimdeblauwe** [reply]
Looks great. Would be happy to have this as a separate library. I am also thinking about a library that would have some helpers for Spring Boot. Not sure if I would combine this into a `htmx-spring-boot-thymeleaf` library or have separate `htmx-thymeleaf` and `htmx-spring-boot` libraries.

**Reply by checketts** [reply]
I'm fine with either approach. I suspect I'll only be using Thymeleaf/HTMX in conjunction with Spring Boot. However I could see some other HTMX helpers (like header helpers or Kotlin extensions) that wouldn't require Thymeleaf.

I have contributed to libraries often, but haven't hosted any.
